From 28e48a2b0b1b50a4dc09051f1250ae667614bedb Mon Sep 17 00:00:00 2001
From: tilghman <tilghman@614ede4d-c843-0410-af14-a771ab80d22e>
Date: Tue, 29 Jul 2008 23:00:45 +0000
Subject: [PATCH] Minor changes to reduce packaging changes made by the Fedora maintainer.
 (closes issue #12974)
  Reported by: jcollie
  Patches:
        0001-Don-t-override-duplicate-optimization-flags.patch uploaded by jcollie (license 412)

git-svn-id: http://svn.digium.com/svn/asterisk/trunk@134312 614ede4d-c843-0410-af14-a771ab80d22e
---
 Makefile |   16 ++++++++++++----
 1 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index a874a7c..f82602c 100644
--- a/Makefile
+++ b/Makefile
@@ -221,10 +221,16 @@ ifeq ($(OSARCH),linux-gnu)
 endif
 
 ifeq ($(findstring -save-temps,$(ASTCFLAGS)),)
-ASTCFLAGS+=-pipe
+  ifeq ($(findstring -pipe,$(ASTCFLAGS)),)
+    ASTCFLAGS+=-pipe
+  endif
+endif
+
+ifeq ($(findstring -Wall,$(ASTCFLAGS)),)
+  ASTCFLAGS+=-Wall
 endif
 
-ASTCFLAGS+=-Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations $(DEBUG)
+ASTCFLAGS+=-Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations $(DEBUG)
 
 ASTCFLAGS+=-include $(ASTTOPDIR)/include/asterisk/autoconfig.h
 
@@ -237,8 +243,10 @@ ifneq ($(findstring BSD,$(OSARCH)),)
   ASTLDFLAGS+=-L/usr/local/lib
 endif
 
-ifneq ($(PROC),ultrasparc)
-  ASTCFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
+ifeq ($(findstring -march,$(ASTCFLAGS)),)
+  ifneq ($(PROC),ultrasparc)
+    ASTCFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
+  endif
 endif
 
 ifeq ($(PROC),ppc)
-- 
1.5.6.5

